projects/storefrontlib/src/cms-structure/services/cms-components.service.ts

```diff
@@ -154,14 +155,44 @@ export class CmsComponentsService {
   /**
    * Get cms driven child routes for components
    */
-  getChildRoutes(componentTypes: string[]): Route[] {
-    const routes = [];
+  getChildRoutes(componentTypes: string[]): CmsComponentChildRoutesConfig {
+    const configs = [];
     for (const componentType of componentTypes) {
       if (this.shouldRender(componentType)) {
-        routes.push(...(this.getMapping(componentType)?.childRoutes ?? []));
+        configs.push(this.getMapping(componentType)?.childRoutes ?? []);
       }
     }
-    return routes;
+
+    return this.standardizeChildRoutes(configs);
+  }
+
+  /**
+   * Standardizes the format to an object with defined parent and children routes, even if parent is undefined
+   *
+   * Some `childRoutes` configs are simple arrays of Routes (without the notion of the parent route).
+   * But some configs can be an object with children routes and their parent defined in separate property.
+   */
+  private standardizeChildRoutes(
+    childRoutesConfigs: (Route[] | CmsComponentChildRoutesConfig)[]
+  ): CmsComponentChildRoutesConfig {
+    return (childRoutesConfigs || []).reduce<CmsComponentChildRoutesConfig>(
+      (result, config) =>
+        Array.isArray(config)
+          ? // config is an array of child routes:
+            {
+              parent: result.parent,
+              children: [...result.children, ...config],
+            }
+          : // config is an object with `parent` and `children` properties:
+            {
+              parent: config.parent ?? result.parent,
+              children: [...result.children, ...config.children],
+            },
+      {
+        parent: undefined,
```

Comment:
I'd prefer to come up with an object without the parent if it is `undefined`. For the 3 cases in this code.

Reply:
I find the code hard to read. 
Perhaps we create the vars, and then put it (optionally, if truthy) in an object?